robot/rospackages/src/mcu_control/scripts/NavigationNode.py

```diff
@@ -19,56 +19,69 @@ def subscriber_callback(message):
         rover['heading'] = message.z
         hasHeading = True
 
-    if gotGpsPos:
-        Rov_to_des_distance = Distance(rover['latitude'], rover['longitude'], \
-        gpsGoal['latitude'], gpsGoal['longitude'])
-        Rov_to_des_direction = Direction(rover['latitude'], rover['longitude'], \
-        gpsGoal['latitude'], gpsGoal['longitude'])
+    if rospy.get_param('has_gps_goal'):
+        rov_to_des_distance = Distance(rover['latitude'], rover['longitude'],
+                                       gpsGoal['latitude'], gpsGoal['longitude'])
+        rov_to_des_direction = Direction(rover['latitude'], rover['longitude'],
+                                         gpsGoal['latitude'], gpsGoal['longitude'])
 
         if hasHeading:
-            Direction_adjust = Turning(Rov_to_des_direction, rover['heading'])
+            direction_adjust = Turning(rov_to_des_direction, rover['heading'])
         else:
-            Direction_adjust = -999 # no heading, give invalid number
+            direction_adjust = -999  # no heading, give invalid number
 
         msg = Point()
-        # note that direction is based on compass directions where E is 90 and W is -90
-        msg.x = Direction_adjust
-        msg.y = Rov_to_des_distance
+        msg.x = direction_adjust # note that direction is based on compass directions where E is 90 and W is -90
+        msg.y = rov_to_des_distance # this value is in meters
+
+        if(msg.y < 10):
+            rospy.loginfo('moving on to next coordinates')
+
+        rospy.loginfo('distance : ' + str(msg.y))
         navigationPub.publish(msg)
     else:
-        rospy.loginfo('Waiting for gps goal coordinate ROS parameters to be set')
+        rospy.loginfo(
+            'Waiting for gps goal coordinate ROS parameters to be set')
     return
 
+
 if __name__ == '__main__':
     node_name = 'navigation_node'
-    rospy.init_node(node_name, anonymous=False) # only allow one node of this type
-    rospy.loginfo('Initialized "'+node_name+'" node for pub/sub functionality')
+    # only allow one node of this type
+    rospy.init_node(node_name, anonymous=False)
+    rospy.loginfo('Initialized "' + node_name +
+                  '" node for pub/sub functionality')
 
     subscribe_topic = '/rover_position'
-    rospy.loginfo('Beginning to subscribe to "'+subscribe_topic+'" topic')
+    rospy.loginfo('Beginning to subscribe to "' + subscribe_topic + '" topic')
     sub = rospy.Subscriber(subscribe_topic, Point, subscriber_callback)
 
     navigation_pub_topic = '/rover_goal'
-    rospy.loginfo('Beginning to publish to "'+navigation_pub_topic+'" topic')
+    rospy.loginfo('Beginning to publish to "' +
+                  navigation_pub_topic + '" topic')
     navigationPub = rospy.Publisher(navigation_pub_topic, Point, queue_size=10)
 
-    gotGpsPos = False
-    gpsGoal = {'latitude':None, 'longitude':None}
-    rover = {'latitude':None, 'longitude':None, 'heading':None, 'distance':None}
+    #gotGpsPos = False
```

Comment:
![delet_this_catto](https://user-images.githubusercontent.com/19865282/84455814-50103780-ac2c-11ea-9979-e16788fbacfe.png)

Reply:
(or uncomment it if it's necessary)

Reply:
I don't think it's needed so I deleted it